src/sql/workbench/services/queryEditor/browser/queryEditorService.ts

```diff
@@ -92,6 +92,11 @@ export class QueryEditorService implements IQueryEditorService {
 		if (sqlContent) {
 			m.textEditorModel.setValue(sqlContent);
 		}
+		m.onDidChangeDirty(() => {
```

Comment:
Actually I was looking into this further and we already have an EditDataInput that even already hooks into the onDidChangeDirty. So you should be able to just add the setDirty call to that instead which is a much better solution IMO.

Reply:
Add a setDirty call to the model in queryEditorService? There is no setDirty call in editDataInput.

Reply:
No, you should be able to get the model from calling 
```
const model = await this._sql.resolve();
model.setDirty(false);
```

Reply:
The sqlContent thing causes the isDirty boolean to be set true due to the change, so it must be hooked in before the textEditorModel is called.